wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -279,6 +279,9 @@ public void delete(@ApiParam("The workspace id") @PathParam("id") String id) thr
                                                                                         ConflictException,
                                                                                         ForbiddenException {
         ensureUserIsWorkspaceOwner(id);
+        if (!workspaceManager.getSnapshot(id).isEmpty()) {
```

Comment:
I understand that it is out scope of current issue. But maybe it would be better to rename getSnapshot to getSnapshots?